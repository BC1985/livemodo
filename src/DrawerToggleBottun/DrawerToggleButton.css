#toggle-button {
  margin: 0 10px 0 10px;
  background: transparent;
  border: none;
  cursor: pointer;
  z-index: 100;
}

#toggle_button:focus {
  outline: none;
}
