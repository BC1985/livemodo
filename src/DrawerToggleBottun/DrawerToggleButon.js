import React, { Component } from "react";
import "./DrawerToggleButton.css";

class drawerToggleButton extends Component {
  render() {
    const { click, isSideDrawerOpen } = this.props;
    const styles = {
      line: {
        height: "3px",
        width: "20px",
        background: "white",
        transition: "all 0.3s ease",
        zIndex: 10
      },
      lineTop: {
        transform: isSideDrawerOpen ? "rotate(50deg)" : "none",
        transformOrigin: "top left",
        marginBottom: "5px"
      },
      lineMiddle: {
        opacity: isSideDrawerOpen ? 0 : 1,
        transform: isSideDrawerOpen ? "translateX(-16px)" : "none"
      },
      lineBottom: {
        transform: isSideDrawerOpen
          ? "translateX(-1px) rotate(-50deg)"
          : "none",
        transformOrigin: "top left",
        marginTop: "5px"
      }
    };
    return (
      <button id="toggle-button" onClick={click}>
        <div style={{ ...styles.line, ...styles.lineTop }} />
        <div style={{ ...styles.line, ...styles.lineMiddle }} />
        <div style={{ ...styles.line, ...styles.lineBottom }} />
      </button>
    );
  }
}

export default drawerToggleButton;
