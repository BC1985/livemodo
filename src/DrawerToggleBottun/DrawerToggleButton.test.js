import React from "react";
import ReactDOM from "react-dom";
import DrawerToggleBottun from "../DrawerToggleBottun/DrawerToggleButon";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DrawerToggleBottun />, div);
  ReactDOM.unmountComponentAtNode(div);
});
