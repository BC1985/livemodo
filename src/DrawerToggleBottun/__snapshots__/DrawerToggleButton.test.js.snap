// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<button
  className="toggle_button"
>
  <div
    className="toggle_button_line"
  />
  <div
    className="toggle_button_line"
  />
  <div
    className="toggle_button_line"
  />
</button>
`;
